services/provider/server/server.go

```diff
@@ -681,12 +707,31 @@ func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.S
 				"clusterID":          getSubVolumeGroupClusterID(subVolumeGroup),
 				"subvolumegroupname": subVolumeGroup.Name,
 				"fsName":             subVolumeGroup.Spec.FilesystemName,
-				"pool":               subVolumeGroup.GetLabels()[v1alpha1.CephFileSystemDataPoolLabel],
+				"pool":               subVolumeGroup.GetLabels()[ocsv1alpha1.CephFileSystemDataPoolLabel],
 				"csi.storage.k8s.io/provisioner-secret-name":       provisionerSecretName,
 				"csi.storage.k8s.io/node-stage-secret-name":        nodeSecretName,
 				"csi.storage.k8s.io/controller-expand-secret-name": provisionerSecretName,
 			}
 
+			clientProfileSpec := &csiopv1a1.ClientProfileSpec{
+				CephFs: &csiopv1a1.CephFsConfigSpec{
+					SubVolumeGroup: cephRes.Name,
+				},
+			}
+			storageClusters := &ocsv1.StorageClusterList{}
+			if err := s.client.List(ctx, storageClusters, client.InNamespace(s.namespace), client.Limit(1)); err != nil {
+				return nil, status.Errorf(codes.Internal, "failed to get storage cluster: %v", err)
+			}
+			if len(storageClusters.Items) < 1 {
+				return nil, status.Errorf(codes.Internal, "storagecluster does not exist")
+			}
+			kernelMountOptions := util.GetCephFSKernelMountOptions(&storageClusters.Items[0])
+			// TODO: do we want to always send the options?
+			if kernelMountOptions == "ms_mode=secure" {
```

Comment:
what happens if the `kernelMountOptions` includes more then "ms_mode"? 
Your check will fail and we will not send it to the client.

Reply:
I could trim but we were getting that string from static code.

Reply:
done.